### PR TITLE
[RELEASE] 4.1.1

### PR DESCRIPTION
**Pull request by JuancaG05**
## Related Issues
App: https://github.com/owncloud/android/issues/4165

_____

### CODE REVIEW

File: owncloudData/src/main/java/com/owncloud/android/data/files/repository/OCFileRepository.kt [hunk 1]
```diff
@@ -168,10 +168,14 @@ class OCFileRepository(
         val personalSpace = localSpacesDataSource.getPersonalSpaceForAccount(owner)
         if (personalSpace == null) {
             val legacyRootFolder = localFileDataSource.getFileByRemotePath(remotePath = ROOT_PATH, owner = owner, spaceId = null)
-            return legacyRootFolder!!
+            try {
+                return legacyRootFolder ?: throw IllegalStateException("LegacyRootFolder not found")
+            } catch (e: IllegalStateException) {
+                Timber.i("There was an error: $e")
```
Comment (manuelplazaspalacio) [inline owncloudData/src/main/java/com/owncloud/android/data/files/repository/OCFileRepository.kt:174]: Placing this catch here could mute the error to the user.

**Review by manuelplazaspalacio — CHANGES_REQUESTED** [review]
One comment here

File: owncloudData/src/main/java/com/owncloud/android/data/files/repository/OCFileRepository.kt (hunk 1 above)
Comment (JuancaG05) [reply owncloudData/src/main/java/com/owncloud/android/data/files/repository/OCFileRepository.kt:174]: This comes from this CR: https://github.com/owncloud/android/pull/4170#discussion_r1338841950
It's true that the user won't receive any feedback, but legacyRootFolder should never be `null`, this allows us to catch the exception and prevent the app from crashing

**Review by manuelplazaspalacio — APPROVED** [review]

**Comment by jesmrec** [comment]
`4.1.1` OK from my side

**Approved commits:** 

app: `cde9bb841`
library: `c82d75b1`

**Android Studio version:** 

`Giraffe | 2022.3.1 Patch 1`

**Reports:**

vs oC 10.3.2: https://reports.cucumber.io/reports/c5cecd13-d2d5-476c-85a6-99ca70a2b4c4
vs oCIS 4.0.0: https://reports.cucumber.io/reports/3e3420a5-b9ac-48bb-ad73-41039968b83e

**Smoke test checklist:** 

https://github.com/owncloud/android/issues/4165#issuecomment-1768877683

**Comments:**

Thumbnails fix is a first approach by updating the cache version that triggers a clean up. Ultimate fix is removing the thumbnails cache by replacing for another library/solution. Should be addressed to a different issue.

**Review by jesmrec — APPROVED** [review]

**Pull request merged**